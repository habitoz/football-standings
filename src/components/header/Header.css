.AppLogo {
  font-family: Montserrat, system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-size: 1.5rem !important;
  font-weight: bold;
  letter-spacing: normal !important;
  color: #fff !important;
  display: flex;
  align-items: center;
  gap: 10px;
}

.Link {
  font-family: Montserrat, system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-size: 1.2rem;
  letter-spacing: 1.9px;
  color: #121212;
  text-decoration: none;
}

.Nav {
  padding: 1rem 6.188rem 1rem 6.25rem;
  background-color: var(--primaryColor);
  color: #fff;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 5rem;
  box-shadow: 0 0 60px 0 rgba(0, 0, 0, 0.05);
}

.MenuIcon {
  display: none;
}

.NavItems {
  display: flex;
  align-items: center;
  gap: 2.563rem;
  list-style: none;
}

@media screen and (width <= 520px) {
  .MenuIcon {
    display: block;
  }
  .NavItems {
    display: none;
  }
}