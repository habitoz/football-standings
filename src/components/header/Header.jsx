import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.css';
import MenuIcon from '../../assets/menu.svg';

export default function Header() {
  return (
    <nav className="Nav">
      <NavLink className="Link AppLogo" to="/">
        <img className="MenuIcon" src={MenuIcon} alt="" />
        <p className="AppName">F standing</p>
      </NavLink>
      <ul className="NavItems">
        <li><NavLink className="Link" to="/">Rockets</NavLink></li>
        <li><NavLink className="Link" to="/missions">Missions</NavLink></li>
        <li><NavLink className="Link" to="/profile">My Profile</NavLink></li>
      </ul>
    </nav>
  );
}
