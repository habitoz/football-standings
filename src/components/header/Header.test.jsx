import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Header from './Header';

const mockStore = configureStore([]);

test('Header component snapshot', () => {
  const initialState = {
    appInfo: {
      activePage: 'League',
      showSearchForm: true,
    },
  };
  const store = mockStore(initialState);
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
        ,
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
