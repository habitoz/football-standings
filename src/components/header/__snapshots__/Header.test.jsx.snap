// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component snapshot 1`] = `
Array [
  <nav
    className="Nav"
  >
    <a
      aria-current="page"
      className="Link AppLogo active"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="MenuIcon"
        src="menu.svg"
      />
    </a>
    <p
      className="AppName"
    >
      League
    </p>
    <button
      className="searchBtn"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-search"
      />
    </button>
    <ul
      className="NavItems"
    />
  </nav>,
  ",",
]
`;
